corehq/apps/case_importer/tracking/views.py

```diff
@@ -118,11 +118,21 @@ def case_upload_form_ids(request, domain, upload_id):
 @conditionally_location_safe(locsafe_imports_enabled)
 def case_upload_case_ids(request, domain, upload_id):
     try:
-        case_upload = CaseUploadRecord.objects.get(upload_id=upload_id, domain=domain)
+        case_upload = _get_case_upload_record(domain, upload_id, request.couch_user)
     except CaseUploadRecord.DoesNotExist:
         return HttpResponseNotFound()
 
     ids_stream = ('{}\n'.format(case_id)
                   for case_id in get_case_ids_for_case_upload(case_upload))
 
     return StreamingHttpResponse(ids_stream, content_type='text/plain')
+
+
+def _get_case_upload_record(domain, upload_id, user):
+    kwargs = {
+        'domain': domain,
+        'upload_id': upload_id,
+    }
+    if not user.has_permission(domain, 'access_all_locations'):
+        kwargs['couch_user_id'] = user._id
+    return CaseUploadRecord.objects.get(**kwargs)
```

Comment:
Do you know how this worked for non-admin users?  It looks like non-admin users are currently restricted from seeing uploads aside from ones they created themselves, in which case you may not need to do anything special for location-restricted users.  It looks like the existing behavior doesn't check in enough places though - does that seem intentional or is it an oversight?  If possible, it would be nice to only have two sets of functionality (admin and non-admin), rather than three (admin, location-restricted, other non-admin).

Reply:
I don't know enough about this to make a judgement call. `tracking.dbaccessords.get_case_upload_records` currently returns all CaseUploadRecords for non-admin users, but they can only view or comment on them if they uploaded them.

Returning all for location-restricted users seems wrong. But returning only their own (which is what this PR does) doesn't seem analogous to the current behaviour for non-admin users. I'd imagine the ideal would be to return all those within their location and below ... but that might not be practical.

Reply:
> I'd imagine the ideal would be to return all those within their location and below ... but that might not be practical.

For sure.  I was hoping non admin users could only see their own uploads, but that sounds like it's only sorta the case.  We might even be able to change that to be in line with the behavior you've implemented here for location-restricted users, but it feels like a product decision and out of scope for these changes.